packages/flutter_tools/lib/src/android/java.dart

```diff
@@ -0,0 +1,245 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:process/process.dart';
+
+import '../base/file_system.dart';
+import '../base/logger.dart';
+import '../base/os.dart';
+import '../base/platform.dart';
+import '../base/process.dart';
+import 'android_studio.dart';
+
+const String _javaHomeEnvironmentVariable = 'JAVA_HOME';
+const String _kJavaExecutable = 'java';
+
+class Java {
+  Java({
+    required this.home,
+    required this.binary,
+    required Logger logger,
+    required FileSystem fileSystem,
+    required OperatingSystemUtils os,
+    required Platform platform,
+    required ProcessManager processManager,
+  }): _logger = logger,
+      _fileSystem = fileSystem,
+      _os = os,
+      _platform = platform,
+      _processManager = processManager,
+      _processUtils = ProcessUtils(processManager: processManager, logger: logger);
+
+  /// Finds the Java runtime that should be used for all java-dependent
```

Comment:
I'm definitely not a Java dev, but would JDK be more accurate than "Java runtime"?

Reply:
A JDK is a superset of a java runtime environment. The naming was also partly inspired the naming of Android Studio's java runtime, [JetBrains Runtime Environment (JBR)](https://github.com/JetBrains/JetBrainsRuntime) (since 2022).

However, now that I think of it, I think every Java-dependent operation we do requires a JDK; so JDK is still correct while also being more precise.

Reply:
JRE is also fine--when I see "runtime" i think a single application, rather than a collection including directory structure.